src/include/containers/List.hpp

```diff
@@ -60,12 +60,17 @@ class List
 {
 public:
 
-	void add(T newNode)
+	bool add(T newNode)
 	{
+		// if newNode already has a sibling don't allow it to be added
+		if (newNode->getSibling() != nullptr) {
```

Comment:
What if it's already added as last item?

Reply:
Checking the list now.

Reply:
Yes it's a bit of an overreaction band-aid to a potential problem (that should now be fixed and likely isn't even present in the current code base). Removed.